### DISCUSSION TITLE
Prisma Raw query failed - Vercel Deployment Error

### DISCUSSION THREAD

**Discussion by RawAnimal** [prisma/prisma#21629] [Q&A]
### Question

@prisma/client:  ^5.3.1 - MySQL - Vercel - Planet Scale

Basically trying to query a table of locations/sites and return the count of how many sites started per week for the last 52 weeks.

**Query**

```
await prisma.$queryRaw`SELECT cast(year(startDate) as CHAR) AS year, week(startDate) AS week, cast(COUNT(*) as CHAR ) AS total FROM site GROUP BY YEAR, WEEK ORDER BY Year DESC, Week DESC LIMIT 52`;
```

**Vercel Error**

```
 Raw query failed. Code: `1146`. Message: `target: esmhq.-.primary: vttablet: rpc error: code = NotFound desc = Table 'esmhq.site' doesn't exist (errno 1146) (sqlstate 42S02) (CallerID: 110a9mhehvkxg0fwucw1): Sql: "select cast(year(startDate) as CHAR) as `year`, week(startDate) as `week`, cast(count(*) as CHAR) as total from site where 1 != 1 group by `YEAR`, `WEEK`", BindVars: {REDACTED}`
```

I'm trying to populate a recharts bar graph. Everything is working fine locally. Just wondering if it something I have done with my query statement. 

Any help that could be provided would be greatly appreciated.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by janpio** [comment]
This is the relevant part of the error message:
```
Table 'esmhq.site' doesn't exist
```

So in the database `esmhq` the table `site` does not exist, which you are trying to access.

**Reply by RawAnimal** [reply]
Hi there. Thanks for the response. 

<img width="541" alt="image" src="https://github.com/prisma/prisma/assets/2855017/26477878-5907-4d98-ada7-7a9f27a95c27">

I thought something had happened to the table as well but it is still there.

I hope the image I added shows. This is the first time I've asked for help so unsure.

**Reply by RawAnimal** [reply]
Is it because the capitalization? Just noticed that. Will check.

**Reply by janpio** [reply]
Yes, that is very possible.
The error comes directly from the _database_, not Prisma, we just rely it. So even if that was not the cause, nothing we could do about it besides to point you in all the other directions that could cause this (double check database connection string and so on).

**Comment by RawAnimal** [comment]
Here's the connection string directly from the code

`const newSites52: object =
      await prisma.$queryRaw`SELECT cast(year(startDate) as CHAR) AS year,
 week(startDate) AS week, cast(COUNT(*) as CHAR ) AS total FROM site
  GROUP BY YEAR, WEEK ORDER BY Year DESC, Week DESC LIMIT 52`;`

it's lower case in the query. Any other ideas?

**Reply by RawAnimal** [reply]
I'll try capitalizing in the query and see if that helps. It works fine locally in dev. Right in the middle of a few changes so can't upload to check immediately. Will do so as soon as I can. I really do appreciate the response. Pulling my hair out with this. lol

**Reply by janpio** [reply]
Note that in some SQL queries, sometimes (I bet there are rules, but I am not aware of them) you have to quote identifiers or they might be modified. Unfortunately also how you do that is different based on the database and so on - so best google a bit for your case.

**Comment by RawAnimal** [comment]
It's been awhile. I was able to overcome the issue I was having with the
Raw Query by using a stored procedure. worked perfectly. Thanks for all
your help.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
